### PR TITLE
add test setup instructions

### PR DESCRIPTION
**Pull request by christothes**
### Fix or Enhancement?
Enhancement to readme.md only

- [ ] All tests passed
- [ ] Add change to `changelog.md`

### Environment
- OS: n/a
- Go version: n/a

### CODE REVIEW

**Comment by devigned** [comment]
Should just need to run `make test`. Terraform will run and create the .env file automatically.

**Comment by christothes** [comment]
> Should just need to run `make test`. Terraform will run and create the .env file automatically.
> 
> Also, Terraform will create all the Azure infrastructure needed for testing.

That definitely sounds simpler! On windows, I ran into some dependency issues that I need to work through to get that working. I'll modify the PR with the details.

**Comment by devigned** [comment]
A while back I started on a set of make like PS cmdlets to help out Windows devs. At this point, I think it might be easier to use the Deis go-dev image on Windows and just build in the container similar to `make dev` as shown here: https://github.com/Azure/aks-engine/blob/38ff602a70533c6305bbc96d2d631bb690c7de81/Makefile#L61

**Comment by christothes** [comment]
I was thinking of just of leveraging WSL, once the dependencies are all installed. Does that sound like less overhead than the container route?

**Comment by devigned** [comment]
Sounds totally reasonable, @christothes. 

I think any opinionated guidance to help Windows developer is welcomed. If the flow feels good for you using WSL, then I'm betting it'll be great for others too.

Thank you for helping to improve our little corner of the interwebs.

**Comment by devigned** [comment]
I noticed there was stuff added to `go.mod` and `go.sum`. I think you want to run `go mod tidy` to clean up the stuff that was added when running `go get ...` from the root of the repo directory.

**Comment by christothes** [comment]
> I noticed there was stuff added to `go.mod` and `go.sum`. I think you want to run `go mod tidy` to clean up the stuff that was added when running `go get ...` from the root of the repo directory.

I appreciate the guidance!
One question I had though - is it expected that I had to install dependencies such as gocyclo as opposed to them being automatically brought in by `make test`? I wasn't sure if these mod and sum changes should be part of the PR because of the those packages being needed anyway. Sounds like I need to look further into how dependency management is supposed to work here :)

**Comment by devigned** [comment]
Awesome question! This brings up one of the uglier parts of go modules (currently). When you need dev / test tools, you don't have dev dependencies like you do in some other languages.

Generally, folks solve it by creating a `/tools/` or `/hack/tools/` dir which will contain a `tools` module and bin directory to hold those tools. A good example of the tools pattern is in [apmz](https://github.com/devigned/apmz) where the `./tools/` directory has it's own module and the [Makefile](https://github.com/devigned/apmz/blob/3d0212e023b9330d7364dfac0ef94308ff39688d/Makefile#L21-L23) calls tools makefile to install the tools into `./tools/bin/`. From there, the top level [makefile then uses the project local in directory for build tools](https://github.com/devigned/apmz/blob/3d0212e023b9330d7364dfac0ef94308ff39688d/Makefile#L11-L13). The result is that all of the bins needed for builds are self-contained and easily installed without polluting the top level `go.mod` and `go.sum`.

When I first built out this project, I wasn't clear on the best way to handle this, and just relied on having `go mod tidy` in the makefile to clean up the local cruft before pushing changes remotely.

**Comment by christothes** [comment]
Thanks for the great explanation and references!

I decided to defer the dev tools refactoring for some future PR and just tidy along with adding the additional dependencies to the readme. Does it look correct now?

**Comment by devigned** [comment]
Good call to break those PRs up. LGTM

File: readme.md [hunk 1]
```diff
@@ -434,6 +434,30 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+- [install WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10) (if on Windows)
+- [install golang](https://golang.org/doc/install) 
+- add paths to .profile
+  - export PATH=$PATH:/usr/local/go/bin:$HOME/go/bin
+  - export GOPATH=$HOME/go
+- install go dev dependencies
+  - run `go get github.com/fzipp/gocyclo`
+  - run `go get -u golang.org/x/lint/golint`
+- run the following bash commands
+  - `sudo apt-get install jq`
```
Comment (gavinfish) [inline readme.md]: nit: use `apt` instead of `apt-get` to be consistent with below commands.

File: readme.md [hunk 2]
```diff
@@ -434,6 +434,30 @@ contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additio
...
+- install gcc
+  - on Ubuntun:
```
Comment (gavinfish) [inline readme.md]: typo `Ubuntun`

**Comment by christothes** [comment]
@gavinfish - Does everything look ok to merge?

**Pull request merged**